cmake/BuildParameters.cmake

```diff
@@ -349,6 +351,19 @@ else()
     set(LTO_FLAGS "")
 endif()
 
+if (USE_PGO_GENERATE)
+    message(WARNING "PGO has not been thoroughly tested. A bug is not impossible.")
+    set(PGO_FLAGS "-fprofile-dir=${CMAKE_SOURCE_DIR}/profile -fprofile-generate")
+    if(USE_PGO_OPTIMIZE)
+        message(WARNING "USE_PGO_OPTIMIZE and USE_PGO_GENERATE should not be used together! USE_PGO_GENERATE will most likely remove any benefits gained by USE_PGO_OPTIMIZE, and the emulator will run slowly!")
```

Comment:
use a fatal error (check the syntax) and stop at `together`. You could remove the set below. Actually I think it will be better to move the check above. It feels easier to read.
```
if (opt and gen)
   fatal error
endif
# then the handling
if()
elseif()
else()
endif()
```